agent-service/tests/test_debate_feature.py

```
@@ -146,3 +140,35 @@ async def generate(self, messages, temperature: float = 0.4, max_tokens: int = 6
+            }
+
+    monkeypatch.setattr(
+        debate_feature_module,
+        "ensure_retriever",
+        lambda *_args, **_kwargs: (_Retriever(), None),
+    )
+    monkeypatch.setattr(LLMProviderFactory, "create", lambda: _StubLLM())
+
+    with pytest.raises(
+        ValueError,
+        match="Roster Agent returned an invalid structured response.",
```

Ruff (0.15.9)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)